applications/graphql-lambda/main.tf

```diff
@@ -0,0 +1,72 @@
+# Experimental: This module builds a serverless GraphQL instance.
+
+variable "environment" {
+  description = "The environment for this application: development, qa, or production."
+}
+
+variable "name" {
+  description = "The application name."
+}
+
+variable "papertrail_destination" {
+  description = "The Papertrail log destination for this application."
+}
+
+locals {
+  safe_name = "${replace(var.name, "-", "_")}"
+}
+
+resource "aws_s3_bucket" "deploy" {
+  bucket = "${var.name}-deploy"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_s3_bucket_object" "build" {
+  bucket = "${aws_s3_bucket.deploy.id}"
+  key    = "${local.safe_name}.zip"
+  source = "${file("${path.module}/example.zip")}"
+}
+
+resource "aws_lambda_function" "function" {
+  function_name = "${var.name}"
+  handler       = "main.handler"
+
+  # The bucket name as created earlier with "aws s3api create-bucket"
```

Comment:
@DFurnes So, in this example, you would first need the Lambda build packaged up locally as a zip?

Reply:
Ah, that comment is a lie! My hope is that we can push up a "default" function (stored in `example.zip`) when provisioning the new resource and then manage uploading new releases via a separate IAM "deploy" role without invalidating state.